video/out/cocoa_common.m

```diff
@@ -466,14 +466,21 @@ static CVReturn displayLinkCallback(CVDisplayLinkRef displayLink, const CVTimeSt
     return kCVReturnSuccess;
 }
 
-static void vo_set_level(struct vo *vo, int ontop)
+static void vo_set_level(struct vo *vo, int ontop, int ontop_level)
 {
     struct vo_cocoa_state *s = vo->cocoa;
 
     if (ontop) {
-        // +1 is not enough as that will show the icon layer on top of the
-        // menubar when the application is not frontmost. so use +2
-        s->window_level = NSMainMenuWindowLevel + 2;
+        switch (ontop_level) {
+        case -1:
+            s->window_level = NSFloatingWindowLevel;
+            break;
+        case -2:
+            s->window_level = NSStatusWindowLevel;
+            break;
+        default:
+            s->window_level = ontop_level;
```

Comment:
Not sure about the Cocoa specific code, but for --ontop-level it might actually be a somewhat ok idea. The option could be reused for X11, which has a _WIN_LAYER window property which can be between 0 and 12 (only reference I found: ftp://ftp.math.utah.edu/u/ma/hohn/linux/gnome/gnome-libs/devel-docs/WM-GOME.txt).